0_intro_to_ml_using_jax.ipynb

```diff
@@ -0,0 +1,3758 @@
+{
```

Comment:
<p>Just. tocheck my understanding so that I can answer questions related to this:</p><p>JAX: can get gradient for any python function (may be some limitations, not sure if it needs to be pure, etc)</p><p>TF and PyTorch: we have to use the frameworks functions (torch.relu, etc) if we want it to be able to calculate the grad.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-learning-indaba/indaba-pracs-2022/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='85'/>

Reply:
<p>So TF and PyTorch would have their own autograd things, but jax is less cleaner (e.g. no <code>.backward()</code> in PyTorch or <code>tape.gradient</code> in tf). You can just define a function naturally and get the grad, I also think it can diff over more diverse things like loops, branches, recursion, and closures.   </p>